service/history/shard/context_test.go

```diff
@@ -1207,3 +1210,316 @@ func TestShardClosedGuard(t *testing.T) {
 		})
 	}
 }
+
+// Mock task that implements persistence.Task interface for testing
```

Comment:
We usually generate a mock from corresponding interface via `//go:generate mockgen` directive. It's more reusable and keeping things consistent across tests. In case you need a fake implementation to keep state across func calls etc. then a custom impl. like this may be preferred.